_extensions/multibib/multibib.lua

```diff
@@ -80,30 +106,53 @@ local function resolve_doc_citations (doc)
   doc = citeproc(doc)
   -- remove catch-all bibliography and keep it for future use
   refs_div_with_properties = table.remove(doc.blocks)
-  -- restore bibliography to original value
-  doc.meta.bibliography = orig_bib
+  -- restore bibliography and references to original values
+  doc.meta.bibliography = bibconf
+  doc.meta.references = refconf
   return doc
 end
 
---- Explicitly create a new meta object with all fields relevant for
---- pandoc-citeproc.
-local function meta_for_pandoc_citeproc (bibliography)
+--- Explicitly create a new meta object with all fields relevant for citeproc.
+local function meta_for_citeproc (bibliography, topic)
   -- We could just indiscriminately copy all meta fields, but let's be
   -- explicit about what's important.
   local fields = {
     'bibliography', 'references', 'csl', 'citation-style',
     'link-citations', 'citation-abbreviations', 'lang',
     'suppress-bibliography', 'reference-section-title',
-    'notes-after-punctuation', 'nocite'
+    'notes-after-punctuation', 'nocite', 'link-bibliography'
   }
   local new_meta = pandoc.Meta{}
   for _, field in ipairs(fields) do
-    new_meta[field] = doc_meta[field]
+    local value = doc_meta[field]
+    -- replace 'references' with the topic references
+    if field == 'references' and metatype(value) == 'table' and topic then
+      value = topic_refs[topic]
+    end
+    new_meta[field] = value
   end
   new_meta.bibliography = bibliography
   return new_meta
 end
 
+-- list of ref-xxx identifiers that have already been output
+local identifiers = List()
+
+-- ignore duplicate references (the first definition will win)
+local function ignore_duplicates(blocks)
+  local new_blocks = pandoc.Blocks{}
+  for _, block in ipairs(blocks) do
+    local identifier = block.attr.identifier
+    if not identifiers:includes(identifier) then
+      local new_block = pandoc.walk_block(block, {Span=_span})
```

Comment:
I think this line doesn't do anything, as `_span` is `nil`? Or does this just serve as a backwards-compatible way to `block:clone()`?